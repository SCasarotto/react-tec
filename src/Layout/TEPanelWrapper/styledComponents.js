import styled from 'styled-components'

export const Container = styled.div`
	${(props) => {
		const { theme, wrapperHeight } = props
		return `
			display: flex;
			flex-flow: row wrap;
			justify-content: space-between;
			align-items: flex-start;
			padding: 30px;
			background-color: ${theme.lighterGray};
			min-height: ${wrapperHeight ? `calc(100vh - ${wrapperHeight}px)` : '100%'};

			@media (max-width: 800px) {
				padding: 25px;
			}
			@media (max-width: 650px) {
				padding: 20px;
			}
			@media (max-width: 450px) {
				padding: 15px;
			}
		`
	}}
`
