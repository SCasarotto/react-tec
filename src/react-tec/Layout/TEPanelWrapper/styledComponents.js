import styled from 'styled-components'

export const Container = styled.div`
	${(props) => {
		const { theme } = props
		return `
			display: flex;
			flex-flow: row wrap;
			justify-content: space-between;
			align-items: flex-start;
			padding: 30px;
			background-color: ${theme.lighterGray};

			@media (max-width: 800px) {
				padding: 25px;
			}
			@media (max-width: 650px) {
				padding: 20px;
			}
			@media (max-width: 450px) {
				padding: 15px;
			}
		`
	}}
`
