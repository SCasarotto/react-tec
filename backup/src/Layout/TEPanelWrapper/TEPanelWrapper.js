import React from 'react'

import { Container } from './styledComponents'

const TEPannelWrapper = (props) => {
	const { className = '', ...rest } = props
	return <Container className={`TEPanelWrapper ${className}`} {...rest} />
}

export default TEPannelWrapper
