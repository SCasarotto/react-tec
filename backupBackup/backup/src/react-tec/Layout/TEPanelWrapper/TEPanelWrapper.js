//
//TE Version 0.3.0
//

import React from 'react'

import { Container } from './styledComponents'

const TEPannelWrapper = (props) => {
	const { className, children } = props
	return <Container className={className}>{children}</Container>
}

export default TEPannelWrapper
