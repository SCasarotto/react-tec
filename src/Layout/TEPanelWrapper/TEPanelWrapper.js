//
//TE Version 0.3.0
//

import React from 'react'

import { Container } from './styledComponents'

const TEPannelWrapper = (props) => <Container {...props} />

export default TEPannelWrapper
