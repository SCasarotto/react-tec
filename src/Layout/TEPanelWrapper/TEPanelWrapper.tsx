import React from 'react'

import { Container } from './styledComponents'

export interface TEPanelWrapperProps extends React.HTMLAttributes<HTMLDivElement> {}
/**
 *
 * TEPanelWrapper simple layout component meant to be used as the parent of `TEPanel`. Below is the code that wraps this page.
 *
 * ```
 * <TEPanelWrapper>
 *		{ ... }
 * </TEPanelWrapper>
 * ```
 *
 */
export const TEPanelWrapper: React.FC<TEPanelWrapperProps> = (props) => {
	const { className = '', ...rest } = props
	return <Container className={`TEPanelWrapper ${className}`} {...rest} />
}
