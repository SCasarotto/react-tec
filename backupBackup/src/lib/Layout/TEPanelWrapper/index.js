import TEPanelWrapper from './TEPanelWrapper'

export default TEPanelWrapper
