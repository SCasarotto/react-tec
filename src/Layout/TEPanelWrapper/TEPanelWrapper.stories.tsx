import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEPanelWrapper, TEPanelWrapperProps } from './TEPanelWrapper'

export default {
	title: 'Layout/TEPanelWrapper',
	component: TEPanelWrapper,
	argTypes: {},
} as Meta

const Template: Story<TEPanelWrapperProps> = (args) => <TEPanelWrapper {...args} />

export const Default = Template.bind({})
Default.args = {}
