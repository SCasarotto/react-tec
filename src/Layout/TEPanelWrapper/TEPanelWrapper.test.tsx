import React from 'react'
import ReactDOM from 'react-dom'
import { TEPanelWrapper } from './TEPanelWrapper'

describe('TEPanelWrapper', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<TEPanelWrapper />, div)
		ReactDOM.unmountComponentAtNode(div)
	})
})
