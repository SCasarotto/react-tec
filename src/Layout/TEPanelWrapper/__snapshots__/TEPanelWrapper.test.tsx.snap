// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEPanelWrapper matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx iPwRjN TEPanelWrapper "
  />
</DocumentFragment>
`;
